presto-main/src/main/java/com/facebook/presto/operator/AggregationOperator.java

```diff
@@ -256,4 +323,57 @@ public long getEstimatedSize()
             return aggregation.getEstimatedSize();
         }
     }
+
+    private static class AggregationOrderProcessor
```

Comment:
Couldn't this be implemented with a decorator? An `Accumulator` that accumulates everything it gets in `addInput` and then sorts, feeds a delegate and produces output in `evaluate`?